ntoskrnl/io/iomgr/driver.c

```diff
@@ -524,7 +524,18 @@ IopInitializeDriverModule(
     RtlZeroMemory(driverObject, ObjectSize);
     driverObject->Type = IO_TYPE_DRIVER;
     driverObject->Size = sizeof(DRIVER_OBJECT);
-    driverObject->Flags = DRVO_LEGACY_DRIVER; // TODO: check the WDM_DRIVER flag on the module
+
+    PIMAGE_NT_HEADERS NtHeaders = RtlImageNtHeader(ModuleObject->DllBase);
+    // NOTE: We suppose that since the driver has been successfully loaded,
+    // its NT and optional headers are all valid and have expected sizes.
+    ASSERT(NtHeaders);
+    ASSERT(ModuleObject->SizeOfImage == NtHeaders->OptionalHeader.SizeOfImage);
+    ASSERT(ModuleObject->EntryPoint == RVA(ModuleObject->DllBase, NtHeaders->OptionalHeader.AddressOfEntryPoint));
```

Comment:
Can you move that before `RtlZeroMemory`? You're in the middle of `driverObject->` sequence here :)